File: python/aes.py
import hashlib
import sys
import Random
from Cipher import AES


def _read_file(filename):
    f = open(filename, "r")
    return f.read()


def _pad(unpadded_text):
    number_of_bytes_to_pad = AES.block_size - len(unpadded_text) % AES.block_size
    ascii_string = chr(number_of_bytes_to_pad)
    padding_str = number_of_bytes_to_pad * ascii_string
    padded_text = unpadded_text + padding_str
    return padded_text


def _unpad(padded_text):
    padding_length = padded_text[len(padded_text) - 1:]
    return padded_text[:-ord(padding_length)]


def _get_sha256_encrypt(secret):
    return hashlib.sha256(secret.encode()).digest()


def encrypt(secret, raw_text):
    key = _get_sha256_encrypt(secret)
    padded_text = _pad(raw_text)
    iv = Random.new().read(AES.block_size)
    cipher = AES.new(key, AES.MODE_CBC, iv)
    ciphertext = cipher.encrypt(padded_text.encode())
    print((iv + ciphertext).hex())


def decrypt(secret, encrypted_text):
    key = _get_sha256_encrypt(secret)
    encrypted_text = bytes.fromhex(encrypted_text)
    iv = encrypted_text[:AES.block_size]
    cipher = AES.new(key, AES.MODE_CBC, iv)
    plain_text = cipher.decrypt(encrypted_text[AES.block_size:]).decode("utf-8")
    print(_unpad(plain_text))


def main(argv):
    secret = _read_file('secret.txt')
    input_text = _read_file('input.txt')
    if len(argv) == 2 and argv[1] == "encrypt":
        encrypt(secret, input_text)
    elif len(argv) == 2 and argv[1] == "decrypt":
        decrypt(secret, input_text)
    else:
        print("Encryption: put raw text in input.txt and secret in secret.txt")
        print("            then 'python aes.py encrypt'")
        print("Decryption: put encrypted text (hex) in input.txt and secret in secret.txt")
        print("            then 'python aes.py decrypt'")


if __name__ == '__main__':
    main(sys.argv)

File: golang/aes.go
package main

import (
	"bytes"
	"crypto/aes"
	"crypto/cipher"
	"crypto/rand"
	"crypto/sha256"
	"encoding/hex"
	"fmt"
	"io/ioutil"
	"os"
)

func _readFile(filename string) string {
	content, err := ioutil.ReadFile(filename)
	if err != nil {
		panic(err)
	}
	if len(content) == 0 {
		panic("Please check content in " + filename)
	}
	return string(content)
}

func _getSha256Encrypt(secret string) []byte {
	h := sha256.New()
	h.Write([]byte(secret))
	return h.Sum(nil)
}

func _pad(unpaddedText string) []byte {
	numberOfBytesToPad := aes.BlockSize - len(unpaddedText)%aes.BlockSize
	paddedText := make([]byte, len(unpaddedText)+numberOfBytesToPad)
	copy(paddedText[:len(unpaddedText)], unpaddedText)
	copy(paddedText[len(unpaddedText):], bytes.Repeat([]byte{byte(rune(numberOfBytesToPad))}, numberOfBytesToPad))
	return paddedText
}

func _unpad(paddedText []byte) []byte {
	paddingLength := int(paddedText[len(paddedText)-1])
	return paddedText[:len(paddedText)-paddingLength]
}

func encrypt(secret string, rawText string) {
	key := _getSha256Encrypt(secret)
	paddedText := _pad(rawText)

	block, err := aes.NewCipher(key)
	if err != nil {
		panic(err)
	}
	ciphertext := make([]byte, aes.BlockSize+len(paddedText))
	_, err = rand.Read(ciphertext)
	if err != nil {
		panic(err)
	}
	iv := ciphertext[:aes.BlockSize]
	mode := cipher.NewCBCEncrypter(block, iv)
	mode.CryptBlocks(ciphertext[aes.BlockSize:], paddedText)

	fmt.Printf("%x\n", ciphertext)
}

func decrypt(secret string, encryptedTxt string) {
	key := _getSha256Encrypt(secret)
	ciphertext, _ := hex.DecodeString(encryptedTxt)

	block, err := aes.NewCipher(key)
	if err != nil {
		panic(err)
	}
	if len(ciphertext) < aes.BlockSize {
		panic("ciphertext too short")
	}
	iv := ciphertext[:aes.BlockSize]
	ciphertext = ciphertext[aes.BlockSize:]
	paddedText := make([]byte, len(ciphertext))
	if len(ciphertext)%aes.BlockSize != 0 {
		panic("ciphertext is not a multiple of the block size")
	}
	mode := cipher.NewCBCDecrypter(block, iv)
	mode.CryptBlocks(paddedText, ciphertext)

	fmt.Printf("%s\n", _unpad(paddedText))
}

func main() {
	secret := _readFile("secret.txt")
	inputText := _readFile("input.txt")
	if len(os.Args) == 2 && os.Args[1] == "encrypt" {
		encrypt(secret, inputText)
	} else if len(os.Args) == 2 && os.Args[1] == "decrypt" {
		decrypt(secret, inputText)
	} else {
		fmt.Println("Invalid Argument.")
	}
}
